### DISCUSSION TITLE
How to encrypt adonis js 5 response data using middleware

### DISCUSSION THREAD

**Discussion by pawan-khade** [adonisjs/core#3771] [Help] [answered]
I want to encrypt data sent from adonis js 5 response using middleware.
I am returning response data from my service class api to react front end in this way:

public async login(username:string,password:string)
{
   return {'status':'success','data':{'username':username,'password':password }};
}

Now I want to get hold of this data in adonis js 5 middleware and first encrypt it and then send to client. I am unable to get hold of returned response data in adonis js 5 middleware. Please help me out.

**Comment by RomainLanz** [comment, marked as answer]
Hey @pawan-khade! 👋🏻 

What have you tried so far?
You should create a downstream middleware to alter the response.

```ts
import type { HttpContextContract } from '@ioc:Adonis/Core/HttpContext'

class EncryptResponseMiddlware {
  public async handle({ request }: HttpContextContract, next: () => Promise<void> ) {
    await next()

	// Your code here
  }
}
```

**Comment by pawan-khade** [comment]
thanks for your reply. It worked for me. Thanks @RomainLanz
